catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptions.java

```diff
@@ -114,13 +128,18 @@ private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
     private Map<String, Metacard> getRegistryMetacardsMap() throws FederationAdminException {
 
         Map<String, Metacard> registryMetacardsMap = new HashMap<>();
-        List<Metacard> registryMetacards = federationAdminService.getRegistryMetacards();
+        try {
+            List<Metacard> registryMetacards = javax.security.auth.Subject.doAs(subject,
+                    (PrivilegedExceptionAction<List<Metacard>>) () -> federationAdminService.getRegistryMetacards());
 
-        for (Metacard metacard : registryMetacards) {
-            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                    .getValue()
-                    .toString();
-            registryMetacardsMap.put(registryId, metacard);
+            for (Metacard metacard : registryMetacards) {
+                String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
```

Comment:
Registry metacards should always have a registry id (kind of like a metacard id).  I was just looking though the code and there isn't anything that strictly enforces that on creation unless it goes through the registry transformer. Probably should add that check to one of our plugins.